app/bundles/EmailBundle/Swiftmailer/Momentum/Service/SwiftMessageService.php

```diff
@@ -0,0 +1,134 @@
+        TranslatorInterface $translator
+    ) {
+        $this->translator = $translator;
+    }
+
+    /**
+     * @param \Swift_Mime_Message $message
+     *
+     * @return TransmissionDTO
+     */
+    public function transformToTransmission(\Swift_Mime_Message $message)
+    {
+        $messageFrom      = $message->getFrom();
+        $messageFromEmail = current(array_keys($messageFrom));
+        $from             = new TransmissionDTO\ContentDTO\FromDTO($messageFromEmail);
+        if (!empty($messageFrom[$messageFromEmail])) {
+            $from->setName($messageFrom[$messageFromEmail]);
+        }
+        $content = new TransmissionDTO\ContentDTO($message->getSubject(), $from);
+
+        $metadataProcessor = new MetadataProcessor($message);
+
+        if ($body = $message->getBody()) {
+            $content->setHtml($body);
+        }
+
+        $headers = $message->getHeaders()->getAll();
+        /** @var \Swift_Mime_Header $header */
+        foreach ($headers as $header) {
+            if ($header->getFieldType() == \Swift_Mime_Header::TYPE_TEXT &&
+                !in_array($header->getFieldName(), [
+                'Content-Transfer-Encoding',
+                'MIME-Version',
+                'Subject',
+            ])) {
+                $content->addHeader($header->getFieldName(), $header->getFieldBodyModel());
+            }
+        }
+
+        if ($messageText = PlainTextMassageHelper::getPlainTextFromMessage($message)) {
+            $content->setText($messageText);
+        }
+
+        if ($message instanceof MauticMessage) {
+            foreach ($message->getAttachments() as $attachment) {
+                if (file_exists($attachment['filePath']) && is_readable($attachment['filePath'])) {
+                    try {
+                        $swiftAttachment = \Swift_Attachment::fromPath($attachment['filePath']);
+
+                        if (!empty($attachment['fileName'])) {
+                            $swiftAttachment->setFilename($attachment['fileName']);
+                        }
+
+                        if (!empty($attachment['contentType'])) {
+                            $swiftAttachment->setContentType($attachment['contentType']);
+                        }
+
+                        if (!empty($attachment['inline'])) {
+                            $swiftAttachment->setDisposition('inline');
+                        }
+                        $attachmentContent = $swiftAttachment->getEncoder()->encodeString($swiftAttachment->getBody());
+                        $attachment        = new TransmissionDTO\ContentDTO\AttachementDTO(
+                            $swiftAttachment->getContentType(),
+                            $swiftAttachment->getFilename(),
+                            $attachmentContent
+                        );
+                        $content->addAttachment($attachment);
+                    } catch (\Exception $e) {
+                        error_log($e);
```

Comment:
? I think we need to inject and use Logger class?

Reply:
Added